src/components/PageSizeSelector/usePageSizeSelector.ts

```diff
@@ -0,0 +1,14 @@
+import { RECENT_PAGE_SIZE_KEY } from "constants/index";
+import { useUpdateURLQueryParams } from "hooks";
+
+/** usePageSizeSelector updates the page size query param and saves the page size to local storage */
+const usePageSizeSelector = () => {
+  const updateURLQueryParams = useUpdateURLQueryParams();
+  const setPageSize = (pageSize: number) => {
```

Comment:
Since we usually make a call to analytics when setting the page size, would it make sense for `setPageSize` to take in an optional Analytics object and log the action if one exists? This could help avoid avoid repeating identical "Change Page Size" event names across the codebase.

Reply:
Got it, that makes sense. Looking ahead though, it does feel less than ideal that event names are magic strings used across the codebase. Do you think something like a string enum of all events we use could help with standardization going forward?

Reply:
All of our Analytics events are all [strongly typed](https://github.com/evergreen-ci/spruce/blob/e3b3a20574a54e6457bab69805c6eee9bf694e4b/src/analytics/version/useVersionAnalytics.ts#L12). So they aren't technically "magic strings" since they are all supported by typescripts type safety.
![image](https://user-images.githubusercontent.com/4605522/171749621-85a0190d-18e6-46f9-b6dc-763c80c97475.png)

But standardizing some common ones for our often reused patterns like Table pagination could be useful. I'm totally up to ticket the work.

Reply:
Wow I totally didn't know about that typing 🤯 thanks for sharing!